website_sale_stock_control/models/product.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+from openerp.http import request
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    # Called the same way as v11 website_sale_stock module
+    inventory_availability = fields.Selection([
+        ('never', 'Always allow to buy products'),
+        ('always', "Don't allow to buy product without stock"),
+    ], string='Website Stock Control', default='never',
+    )
+    # To be inherit by other modules
+    website_qty_available = fields.Float(
+        compute='_compute_website_qty_available',
+        readonly=True)
+
+    @api.multi
+    def _compute_website_qty_available(self):
+        for product in self:
+            product.website_qty_available = product.sudo().with_context(
+                force_company=request.env.user.company_id.id
+            ).qty_available
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    # To be inherit by other modules
+    website_qty_available = fields.Float(
+        compute='_compute_website_qty_available',
+        readonly=True)
+
+    @api.multi
+    def _compute_website_qty_available(self):
```

Comment:
Isn't this inherited from `product.template` automatically?
If not, do you think it would make sense to add this to an `AbstractModel`, so you can inherit it in both `product.template` and `product.product` and DRY? 🤔

Reply:
Ok. I have made an abstract model